velox/common/memory/MmapAllocator.cpp

```diff
@@ -25,9 +25,20 @@ namespace facebook::velox::memory {
 MmapAllocator::MmapAllocator(const Options& options)
     : kind_(MemoryAllocator::Kind::kMmap),
       useMmapArena_(options.useMmapArena),
+      maxMallocBytes_(options.maxMallocBytes),
       capacity_(bits::roundUp(
-          options.capacity / AllocationTraits::kPageSize,
-          64 * sizeClassSizes_.back())) {
+          options.capacity *
+              (100 -
+               (maxMallocBytes_ == 0 ? 0 : options.smallAllocationReservePct)) /
+              100 / AllocationTraits::kPageSize,
+          64 * sizeClassSizes_.back())),
+      mallocReservedBytes_(
+          maxMallocBytes_ == 0
+              ? 0
+              : (capacity_ * AllocationTraits::kPageSize < options.capacity
```

Comment:
```
std::max<int64_t>(0, options.capacity - AllocationTraits::pageBytes(capacity_))
```

Reply:
You cannot subtract 2 uint64_t. It will overflow to a big positive integer.